lib/textarea/textarea.tsx

```diff
@@ -0,0 +1,274 @@
+        }
+
+        this.setState({
+          height,
+          overflow:
+            maxHeight && textarea.scrollHeight + 1 < maxHeight
+              ? 'hidden'
+              : 'auto'
+        });
+      }
+    );
+  };
+
+  showMessage = (
+    message: string,
+    maxLength: number,
+    remaining: number
+  ): string => {
+    if (message) {
+      return message;
+    }
+
+    if (maxLength) {
+      if (remaining <= 0) {
+        return `You've reached ${maxLength} characters`;
+      }
+      return `${remaining} characters remaining`;
+    }
+
+    return '';
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const target: HTMLTextAreaElement = event.target;
+
+    this.setState({
+      remaining: this.calculateRemaining(target.value),
+      value: target.value
+    });
+
+    this.resize();
+    this.props.onChange(target.value, this.props.name);
+  };
+
+  render() {
+    const { props, state, id, textareaRef, handleChange } = this;
+
+    const modifiers = remove(['error', 'success'], props.modifiers);
+
+    const bem = BEM('textarea');
+    bem.addModifiers(modifiers);
+    bem.addModifiers(state.error ? 'error' : '');
+    bem.addModifiers(state.success ? 'success' : '');
+    bem.addClassNames(props.className);
+
+    const browser = BrowserDetect();
+
+    const borderHeight = browser.name === 'firefox' ? 0 : 1;
+
+    return (
+      <div
+        className={bem.getResult()}
+        style={
+          state.height
+            ? { height: `${state.height + 40}px`, ...props.style }
+            : props.style
+        }>
+        <textarea
+          ref={textareaRef}
+          id={id}
+          className={
+            !state.value
+              ? bem.getElement('input')
+              : `${bem.getElement('input')} active`
+          }
+          name={props.name}
+          disabled={props.disabled}
+          readOnly={props.readOnly}
+          required={props.required}
+          autoComplete={props.autoComplete}
+          autoFocus={props.autoFocus}
+          maxLength={
+            props.maxLength && props.maxLength > 0 ? props.maxLength : null
+          }
+          placeholder={props.placeholder}
+          spellCheck={props.spellCheck}
+          value={state.value}
+          rows={1}
+          tabIndex={!props.readOnly && !props.disabled ? 0 : -1}
+          style={{
+            height: state.height ? `${state.height}px` : null,
+            overflow: state.overflow === 'auto' ? 'auto' : null
+          }}
+          // Grammarly control
+          data-gramm={!props.grammarly ? false : null}
+          onChange={handleChange}
+        />
+        <label className={bem.getElement('label')}>{props.label}</label>
+        {!props.readOnly && !props.disabled && (
+          <div
+            className={bem.getElement('border')}
+            style={
+              state.height
+                ? { top: `${state.height + (16 - borderHeight)}px` }
```

Comment:
@DanMad @andreamalvetta for further improvements:
Is this twice the estimate height of a line? If so, we can have smt like 2 * TXT_AREA_LINE_HEIGHT and use it on the places we use this estimate to calculate other heights like in lines 146, 148, 150:

```
	  if (windowWidth === 'small') {
            maxHeight = 200;
          } else if (windowWidth === 'medium') {
            maxHeight = 264;
          } else {
            maxHeight = 392;
          }
```
we could have, for example
```
numberOfLines = 25;
maxHeight = numberOfLines * TXT_AREA_LINE_HEIGHT;
```